helix-view/src/gutter.rs

```diff
@@ -36,6 +36,35 @@ pub fn diagnostic<'doc>(
     })
 }
 
+pub fn git_diff<'doc>(
+    doc: &'doc Document,
+    _view: &View,
+    _theme: &Theme,
+    _config: &Config,
+    _is_focused: bool,
+    _width: usize,
+) -> GutterFn<'doc> {
+    Box::new(move |line: usize, _selected: bool, out: &mut String| {
+        let diff = match doc.line_diffs().get(&(line + 1)) {
+            None => return None,
+            Some(d) => d,
+        };
+
+        let (icon, style) = match diff {
+            helix_vcs::LineDiff::Added => ("│", Style::default().fg(crate::graphics::Color::Green)),
+            helix_vcs::LineDiff::Deleted => ("▸", Style::default().fg(crate::graphics::Color::Red)),
```

Comment:
I think this is not obvious that the current line is being deleted.

Reply:
Agreed, let's use the symbols I proposed in the other PR

Reply:
I used the right aligned version of all the symbols to have enough space between the diagnostic symbols and git signs:

![Screenshot_2022-02-07_10-41-16](https://user-images.githubusercontent.com/23398472/152728056-153e9fce-6d7a-420c-bdd2-4572a59f740e.png)

Reply:
What if we place it left aligned in that 1 char buffer between the line number and text?

Reply:
I tried some variations of what you suggested:

| Left aligned thick bar | Right aligned thick bar | Center aligned thin bar |
| --- | --- | --- |
| ![thick bar left](https://user-images.githubusercontent.com/23398472/152744862-7a98122e-5844-459c-a743-6fea26d31b53.png) | ![thick bar right](https://user-images.githubusercontent.com/23398472/152744905-bf0b5f39-b3ca-42f6-9ac5-49cc54d6f80f.png) | ![thin bar normal](https://user-images.githubusercontent.com/23398472/152744936-eee735f7-3a63-4550-a95c-441bdd88c482.png) |

It looks weird when horizontally scrolling though, so I would suggest sticking with making it a normal gutter component:

![horizontalscrol](https://user-images.githubusercontent.com/23398472/152745331-8a25bee9-2f47-4ff8-af5b-a19839f21064.png)

Reply:
A thin bar could work. We can make the space between the text and the gutter a gutter component, it's just currently hardcoded. Then `gutters = [diagnostics, line_number, git]`

Reply:
You can try more blocks from https://en.wikipedia.org/wiki/Block_Elements#Character_table

"Left one quarter block" or "Left three eights" seem most suitable

Reply:
This would be sort of similar to how VSCode draws it ![](https://assets.digitalocean.com/articles/git-integration-in-visual-studio-code/7.png)

I like it because it uses the existing gap instead of growing the view gutter space by another character

Reply:
On the other hand different terminals handle box drawing characters differently, so for example while the centered thin line looks okay on kitty, it look too thin on wezterm.